### PR TITLE
Add Delete All button to ItemList editor

### LINKED ISSUE #1975: Add a clear button to the item list editor
### Describe the project you are working on
I'm working on an RPG, and I had a traits list that I was just going to hard code in all the possible traits that affect certain features of the game into an OptionButton.

### Describe the problem or limitation you are having in your project
I ended up deciding that was a bad idea and not expandable by addons for my game, so I decided to clear out all the itemlists. However, I have to keep clicking the text box of an item, followed by the delete button at the top of the item list editor, and that gets really tedious if you have a whole lot of items for each item list.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
I think there should be a clear button in the item list editor so you can quickly remove every item for an OptionButton or similar. Maybe even a dialogue confirming that you would like to do that so it's idiot proof.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
concept

I just copy-pasted the delete icon to the clear button, but you can definitely change that.

I didn't mess with code, I just did a little bit of image editing to get that

### If this enhancement will not be used often, can it be worked around with a few lines of script?
I'm not sure, I guess you could write a tool script or something, but I have zero experience with them. You could also just... delete the OptionButton or something, but I feel like adding this would be pretty useful so you don't have to keep deleting/re-adding the OptionButton if you keep changing things.

### Is there a reason why this should be core and not an add-on in the asset library?
It's one button, and I feel that it could be quite useful

### PR DESCRIPTION
**Pull request by KoBeWi**
Closes https://github.com/godotengine/godot-proposals/issues/1975
chUdP56gyn

Loosely depends on #33300, which loosely depends on #43872

### CODE REVIEW

File: editor/plugins/item_list_editor_plugin.cpp [hunk 1]
```diff
@@ -350,6 +357,11 @@ ItemListEditor::ItemListEditor() {
 
 	hbc->add_spacer();
 
+	clear_button = memnew(Button);
```
Comment (EricEzaM) [inline editor/plugins/item_list_editor_plugin.cpp:360]: Should the name in the code match the name given in the display? For consistenct?

`delete_all_button`

**Comment by akien-mga** [comment]
Maybe there should be a confirmation dialog mentioning how many entries will be removed?

File: editor/plugins/item_list_editor_plugin.cpp (hunk 1 above)
Comment (KoBeWi) [reply editor/plugins/item_list_editor_plugin.cpp:360]: Maybe. I thought it's easier to distinguish this way.

**Comment by KoBeWi** [comment]
I could add the dialog, but IMO it would be better if we could just undo it.

**Review by Calinou — APPROVED** [review]
The implementation looks good to me, but this shouldn't be merged until undo/redo support for the ItemList editor is merged (or a confirmation dialog is added).

Adding a confirmation dialog will also make this cherry-pickable to the `3.x` branch.

**Comment by reduz** [comment]
I think the plan is now to refactor the itemlist and other similar nodes to be edited directly on the inspector with the new upcoming PR from @groud. As such, I think it may make more sense to close this? What do you think?

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
> As such, I think it may make more sense to close this? What do you think?

I think it doesn't hurt to merge this in the meantime, if this is refactored to be integrated in the Inspector, the whole file will be removed/redone anyway.

**Comment by akien-mga** [comment]
Needs a rebase.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**